context/app/static/js/components/home-revision/EntityCounts/EntityCounts.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { AppContext } from 'js/components/Providers';
+
+import EntityCount from 'js/components/home-revision/EntityCount';
+import { DatasetIcon, SampleIcon, DonorIcon, CollectionIcon } from 'js/shared-styles/icons';
+import useSearchData from 'js/hooks/useSearchData';
+import { Background, FlexContainer } from './style';
+
+const entityCountsQuery = {
+  size: 0,
+  aggs: { entity_type: { terms: { field: 'entity_type.keyword' } } },
+};
+
+function EntityCounts() {
+  const { elasticsearchEndpoint, nexusToken } = useContext(AppContext);
+  const [entityCounts, setEntityCountsData] = useState(undefined);
+  const { searchData: elasticsearchData } = useSearchData(entityCountsQuery, elasticsearchEndpoint, nexusToken);
+  useEffect(() => {
+    if (Object.keys(elasticsearchData).length) {
+      const entityCountsObject = elasticsearchData.aggregations.entity_type.buckets.reduce((acc, entity) => {
+        const accCopy = acc;
+        accCopy[entity.key] = entity.doc_count;
+        return accCopy;
+      }, {});
+      setEntityCountsData(entityCountsObject);
+    }
+  }, [elasticsearchData]);
+
+  return (
+    <Background>
+      <FlexContainer>
+        <EntityCount icon={DonorIcon} count={entityCounts?.Donor} label="Donors" href="/search?entity_type[0]=Donor" />
+        <EntityCount
+          icon={SampleIcon}
+          count={entityCounts?.Sample}
+          label="Samples"
+          href="/search?entity_type[0]=Sample"
+        />
+        <EntityCount
+          icon={DatasetIcon}
+          count={entityCounts?.Dataset}
+          label="Datasets"
+          href="/search?entity_type[0]=Dataset"
+        />
+        <EntityCount icon={CollectionIcon} count={entityCounts?.Collection} label="Collections" href="/collections" />
```

Comment:
Not sure if this is cleaner (take it or leave it) but you could make this into a `[].map` to make it less copy-and-paste-y:
```suggestion
        [
        [DonorIcon, entityCounts?.Donor, "Donors", "/search?entity_type[0]=Donor"],
       [SampleIcon,entityCounts?.Sample,"Samples","/search?entity_type[0]=Sample"],         
     [DatasetIcon,entityCounts?.Dataset,"Datasets","/search?entity_type[0]=Dataset"],
     [CollectionIcon, entityCounts?.Collection,"Collections","/collections"]
       ].map(([icon, count, label, href]) => (
       <EntityCount
          icon={icon}
          count={count}
          label={label}
          href={href}
        />
       ));
```

Reply:
Thanks, I'll do something similar.